### PR TITLE
[Security Solution] Add `create_index` to `kibana_system` role for Elastic Defend indices

### PR DESCRIPTION
**Pull request by paul-tavares**
## Description

Adds `create_index` privilege to the `kibana_system` role for the following indices:

- `.logs-endpoint.heartbeat-*`
- `.logs-endpoint.diagnostic.collection-*`
- `.logs-endpoint.action.responses-*`

This change is required in support of restricting DOT indexes from being created by non-privileged users. Up until now, when the Elastic Defend package is installed via Fleet, the Datastream templates for a few DOT prefixed indices are installed. The indexes are then created the first time the Elastic Defend integration executable (running on host machines) streams it first document - which under the restriction (in serverless) the Elastic Defend executable will no longer be given privileges to do. The alternative is to have Kibana create the indexes prior to the endpoint streaming any documents to elastic search.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/KibanaOwnedReservedRoleDescriptors.java [hunk 1]
```diff
@@ -152,8 +152,11 @@ static RoleDescriptor kibanaSystem(String name) {
                 // Data telemetry reads mappings, metadata and stats of indices
                 RoleDescriptor.IndicesPrivileges.builder().indices("*").privileges("view_index_metadata", "monitor").build(),
                 // Endpoint diagnostic information. Kibana reads from these indices to send
-                // telemetry
-                RoleDescriptor.IndicesPrivileges.builder().indices(".logs-endpoint.diagnostic.collection-*").privileges("read").build(),
+                // telemetry and also creates the index when policies are first created
+                RoleDescriptor.IndicesPrivileges.builder()
```
Comment (kc13greiner) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/KibanaOwnedReservedRoleDescriptors.java:156]: Heya @paul-tavares 

Question: It looks like the `kibana_system` user has had read access for a bit, can you provide any context as to why it needs to create these indexes now?

Comment (paul-tavares) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/KibanaOwnedReservedRoleDescriptors.java:156]: Thanks @kc13greiner  for looking at this PR. 
This change is necessary in order to address the issues found last week when an attempt was made (and later backed out) to restrict DOT indices for serverless. I have updated the description of this PR to more clearly state the need for the changes, which I will include here now:

> This change is required in support of restricting DOT indexes from being created by non-privileged users. Up until now, when the Elastic Defend package is installed via Fleet, the Datastream templates for a few DOT prefixed indices are installed. The indexes are then created the first time the Elastic Defend integration executable (running on host machines) streams it first document - which under the restriction (in serverless) the Elastic Defend executable will no longer be given privileges to do. The alternative is to have Kibana create the indexes prior to the endpoint streaming any documents to elastic search.

Let me know if you have any other questions... 

_(now: on to figure out whats wrong with the failing test... I may need some help there... 😁 )_

**Comment by paul-tavares** [comment]
Ok. Finally the build is 🟢 . Ready for review

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/KibanaOwnedReservedRoleDescriptors.java (hunk 1 above)
Comment (kc13greiner) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/KibanaOwnedReservedRoleDescriptors.java:156]: Thank you for the information!

**Review by kc13greiner — APPROVED** [review]
In general, we do not like to allow`kibana_system` anything other than `read` on system indices, but there are exceptions! (2/3 of the amended index privileges already have `write` and `auto-configure`). A restriction in serverless is definitely a good reason for an exception.

Thank you for the information - LGTM!

**Comment by paul-tavares** [comment]
Hey @kc13greiner - Do I need to tag this PR with anything so that it:

1. Makes it to the next ES snapshot?
2. Makes it to the next Serverless release?

Also,
Do you know if this changes will show up in Cloud QA tomorrow so that I can test with a live serverless environment? (FYI: I also have @leehinman helping me to enable the DOT prefix restriction in a serverless deployment once I have that created)

thanks for your help and time on this.

**Pull request merged**